core/dbt/contracts/graph/unparsed.py

```diff
@@ -88,7 +88,7 @@ class Docs(dbtClassMixin, Replaceable):
 
 
 @dataclass
-class HasDocs(AdditionalPropertiesMixin, ExtensibleDbtClassMixin, Replaceable):
+class HasColumnProps(AdditionalPropertiesMixin, ExtensibleDbtClassMixin, Replaceable):
```

Comment:
The "access" property should not be in a class that is attached to columns. Probably should be moved into UnparsedNodeUpdate.

Reply:
If you look at ColumnInfo in core/dbt/contracts/graph/nodes.py, the ColumnInfo should be a match for what was HasDocs. Looks like the "docs" field should also be moved out of that class, and into UnparsedNodeUpdate.

Reply:
@gshank When I moved `access` to UnparsedNodeUpdate, it caused a the TestAnalyses::test_postgres_analyses test to break as follows:
```
def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
        # We're not passing the ParsedNodePatch around anymore, so we
        # could possibly skip creating one. Leaving here for now for
        # code consistency.
        patch = ParsedNodePatch(
            name=block.target.name,
            original_file_path=block.target.original_file_path,
            yaml_key=block.target.yaml_key,
            package_name=block.target.package_name,
            description=block.target.description,
            columns=refs.column_info,
            meta=block.target.meta,
            docs=block.target.docs,
            config=block.target.config,
>           access=block.target.access,
        )
E       AttributeError: 'UnparsedAnalysisUpdate' object has no attribute 'access'
```

I was able to fix that by adding a separate `access` property to `UnparsedAnalysisUpdate` but it makes me nervous. What do you think?

Moving docs also breaks a few tests.

Reply:
That should be okay. Yet another sign that we should go in and rearrange those classes...

Reply:
Not sure it makes sense functionally to have `access` on an analysis node given they can't be `ref`'d. Is there an alternative workaround? If not, let's at least leave a comment that `access` isn't actually used to restrict `ref`s for analyses.

Reply:
It's not on the analysis node. It's on the class that's used for pulling in fields from schema files in unparsed,py, which is classes that are used for ferrying things from yaml to nodes (plus a few secondary classes that are used both there and in nodes.py). There is no "access" on the AnalysisNode, and setting it is prevented by code  in the the "patch" method on ParsedNode.

We really need to do some cleanup in unparsed.py.

Reply:
Ahh I see, thank you for the explanation!

Reply:
@gshank I verified that the line you highlighted throws a `mashumaro.exceptions.InvalidFieldValue` exception. Should we catch it here and raise a `ParsingError` exception instead, or do you think we should be validating this somewhere else?